yarn-project/simulator/src/avm/avm_simulator.ts

```diff
@@ -189,7 +186,16 @@ export class AvmSimulator {
       return results;
     } catch (err: any) {
       this.log.verbose('Exceptional halt (revert by something other than REVERT opcode)');
-      if (!(err instanceof AvmExecutionError || err instanceof SideEffectLimitReachedError)) {
+      // FIXME: weird that we have to do this OutOfGasError check because:
+      // 1. OutOfGasError is an AvmExecutionError, so that check should cover both
+      // 2. We should at least be able to do instanceof OutOfGasError instead of checking the constructor name
+      if (
+        !(
+          err.constructor.name == 'OutOfGasError' ||
+          err instanceof AvmExecutionError ||
+          err instanceof SideEffectLimitReachedError
+        )
+      ) {
         this.log.error(`Unknown error thrown by AVM: ${err}`);
         throw err;
```

Comment:
Same issue here as mentioned in the other comment.... Idk why `OutOfGasError` wasn't being caught as part of `AvmExecutionError`, but it wasn't.... AND it was failing `instanceof OutOfGasError`.... Sometimes I hate TS :melting_face: